src/components/NewRuntimeModal.js

```diff
@@ -131,15 +136,22 @@ export class NewRuntimeModalBase extends Component {
 
   getInitialState(runtime, disk) {
     const runtimeConfig = runtime?.runtimeConfig
+    const gpuConfig = runtimeConfig?.gpuConfig
+    const sparkMode = runtimeConfig?.cloudService === cloudServices.DATAPROC ? (runtimeConfig.numberOfWorkers === 0 ? 'master' : 'cluster') : false
+
     return {
       selectedPersistentDiskSize: disk?.size || DEFAULT_DISK_SIZE,
-      sparkMode: runtimeConfig?.cloudService === cloudServices.DATAPROC ? (runtimeConfig.numberOfWorkers === 0 ? 'master' : 'cluster') : false,
+      sparkMode,
       masterMachineType: runtimeConfig?.masterMachineType || runtimeConfig?.machineType,
       masterDiskSize: runtimeConfig?.masterDiskSize || runtimeConfig?.diskSize || DEFAULT_DISK_SIZE,
       numberOfWorkers: runtimeConfig?.numberOfWorkers || 2,
       numberOfPreemptibleWorkers: runtimeConfig?.numberOfPreemptibleWorkers || 0,
       workerMachineType: runtimeConfig?.workerMachineType || defaultDataprocMachineType,
-      workerDiskSize: runtimeConfig?.workerDiskSize || DEFAULT_DISK_SIZE
+      workerDiskSize: runtimeConfig?.workerDiskSize || DEFAULT_DISK_SIZE,
+      gpuEnabled: (!!gpuConfig && !sparkMode) || false,
+      hasGpu: !!gpuConfig,
```

Comment:
We need to behave differently when there's an existing compute with GPUs (more on that later) vs. at compute creation time with no preexisting compute. That's why we have `hasGpu` in addition to `gpuEnabled`.